src/testing/deploy/upgrade_test.go

```diff
@@ -136,32 +143,46 @@ func TestUpgradeTrigger(t *testing.T) {
 					return errors.New("not ready")
 				}
 				return nil
-			})
+			}
+			verifyCommitPost28x := func () error {
+				fmt.Printf("verifyCommitPost28x running with version %s", from)
+				commits, err := c.ListCommit(client.NewRepo(pfs.DefaultProjectName, "TestTrigger2"), nil, nil, 0)
+				require.NoError(t, err)
+				require.Equal(t, 1, len(commits))
+				return nil
+			}
+			verifyCommit := verifyCommitPre28x
+			if semver.Compare("v"+from, "v2.8.0") >= 0 {
+				verifyCommit = verifyCommitPost28x
+			}
+			require.NoErrorWithinTRetry(t, 2*time.Minute, verifyCommit)
 		},
-		func(t *testing.T, ctx context.Context, c *client.APIClient, _ string) { /* postUpgrade */
+		func(t *testing.T, ctx context.Context, c *client.APIClient, from string) { /* postUpgrade */
 			for i := 0; i < 10; i++ {
 				require.NoError(t, c.PutFile(dataCommit, "/hello", strings.NewReader("hello world")))
 			}
 			latestDataCI, err := c.InspectCommit(pfs.DefaultProjectName, dataRepo, "master", "")
 			require.NoError(t, err)
-			require.NoErrorWithinTRetryConstant(t, 5*time.Minute, func() error {
-				ci, err := c.InspectCommit(pfs.DefaultProjectName, "TestTrigger2", "master", "")
-				require.NoError(t, err)
-				aliasCI, err := c.InspectCommit(pfs.DefaultProjectName, dataRepo, "", ci.Commit.Id)
-				if err != nil {
-					return err
-				}
-				if aliasCI.Commit.Id != latestDataCI.Commit.Id {
-					return errors.Errorf("not ready alias commit: %v latest data commit: %v", aliasCI.Commit.Id, latestDataCI.Commit.Id)
-				}
-				return nil
-			}, 10*time.Second)
+			if semver.Compare("v"+from, "v2.8.0") < 0 {
```

Comment:
If we do change verification based on version it should have a comment to explain why. That's the type of thing that will be confusing in a year or two.